File: chapter7/java/designpattern/src/main/java/com/designpattern/TempuraKitchen.java
package com.designpattern;

import java.util.List;

/**
 * Tempura Kitchen will prepare and make Tempura per customer order menu
 *
 * @version 1.0.0
 * @Author Bridget
 * @license MIT
 */
public class TempuraKitchen implements Kitchen {

    /**
     * make tempura food based on customer order items
     *
     * @param items
     */
    @Override
    public void makeFood(List<String> items) {
        if (!items.isEmpty()) {
            System.out.println("make Tempura:[");
            for (String item : items) {
                System.out.println(" {" + item + "} ");
            }
            System.out.println("]");
        }
    }
}

File: chapter7/typescript/TempuraKitchen.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TempuraKitchen = void 0;
/**
 * Tempura Kitchen will prepare and make Tempura per customer order menu
 *
 * @version 1.0.0
 * @Author Bridget
 * @license MIT
 */
class TempuraKitchen {
    /**
     * make tempura food based on customer order items
     *
     * @param items
     */
    makeFood(items) {
        if (items && items.length > 0) {
            console.log("cook Tempura:[");
            for (let item of items) {
                console.log(" {" + item + "} ");
            }
            console.log("]");
        }
    }
}
exports.TempuraKitchen = TempuraKitchen;
